lib/syscall_shim/include/uk/legacy_syscall.h

```diff
@@ -0,0 +1,65 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Authors: Răzvan Vîrtan <virtanrazvan@gmail.com>
+ *
+ *
+ * Copyright (c) 2022, University Politehnica of Bucharest. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef __UK_LEGACY_SYSCALL_H__
+#define __UK_LEGACY_SYSCALL_H__
+
+#define LEGACY_fork
+#define LEGACY_vfork
+#define LEGACY_getpgrp
+#define LEGACY_readlink
+#define LEGACY_symlink
+#define LEGACY_unlink
+#define LEGACY_link
+#define LEGACY_access
+#define LEGACY_open
+#define LEGACY_creat
+#define LEGACY_mkdir
+#define LEGACY_rmdir
+#define LEGACY_mknod
+#define LEGACY_chmod
+#define LEGACY_lchown
+#define LEGACY_chown
+#define LEGACY_rename
+#define LEGACY_lstat
+#define LEGACY_stat
+#define LEGACY_getdents
+#define LEGACY_time
+#define LEGACY_futimesat
```

Comment:
Can you add a comment which syscall should be used instead (at least for the non-obvious), like:
```c
#define LEGACY_SYS_futimesat /* modern: utimensat */
```